### PR TITLE
docstring D205, D400 (one line summary) & D301 (slashes)

### PR DESCRIPTION
**Pull request by peterjc**
This pull request addresses issue #1170

@etal @JoaoRodrigues do the PDB changes make sense?

(Further commits to follow for remaining cases in ``Bio.Phylo`` and ``Bio.SearchIO``)

_Update_: Expanded from D205 to also cover D400 and D301 as well.

### CODE REVIEW

**Comment by peterjc** [comment]
@Bow Are the ``Bio.SearchIO`` docstring changes OK?

@etal, @brandoninvergo @lijax are the ``Bio.Phylo`` changes OK?

Comment welcomed on the other modules docstrings changed too, thanks!

**Comment by peterjc** [comment]
@saketkc could you check the changes to the ``bwa`` doctests please (dealing with unescaped slashes),  thanks?

File: Bio/Sequencing/Applications/_samtools.py [hunk 1]
```diff
@@ -430,10 +429,9 @@ class SamtoolsMergeCommandline(AbstractCommandline):
     >>> from Bio.Sequencing.Applications import SamtoolsMergeCommandline
     >>> out_bam = "/path/to/out_bam"
     >>> in_bam = ["/path/to/input_bam1", "/path/to/input_bam2"]
-    >>> samtools_merge_cmd = SamtoolsMergeCommandline(\
-                                                      out_bam=out_bam,\
-                                                      input_bam=in_bam)
-    >>> print(samtools_merge_cmd)
+    >>> merge_cmd = SamtoolsMergeCommandline(out_bam=out_bam,
+    ...                                               input_bam=in_bam)
```
Comment (saketkc) [inline Bio/Sequencing/Applications/_samtools.py]: Indentation seems off.

File: Bio/PDB/PDBList.py [hunk 2]
```diff
@@ -102,19 +102,22 @@ def __init__(self, server='ftp://ftp.wwpdb.org', pdb=os.getcwd(),
 
     @staticmethod
     def _print_default_format_warning(file_format):
-        """Temporary warning (similar to a deprecation warning) that files are being downloaded in mmCIF"""
+        """Print a warning to stdout.
```
Comment (saketkc) [inline Bio/PDB/PDBList.py]: Should this end with '(PRIVATE)'  like in `Bio/Nexus/Nexus.py` so as to keep this consistent?

File: Bio/Phylo/Consensus.py [hunk 3]
```diff
@@ -384,7 +384,7 @@ def adam_consensus(trees):
 
 
 def _part(clades):
-    """Recursive function of adam consensus algorithm"""
+    """Recursive function of Adam Consensus algorithm."""
```
Comment (saketkc) [inline Bio/Phylo/Consensus.py]: PRIVATE?

File: Bio/Phylo/Consensus.py [hunk 4]
```diff
@@ -443,7 +443,7 @@ def _part(clades):
 
 
 def _sub_clade(clade, term_names):
-    """Extract a compatible subclade that only contains the given terminal names"""
+    """Extract a compatible subclade that only contains the given terminal names."""
```
Comment (saketkc) [inline Bio/Phylo/Consensus.py]: PRIVATE?

File: Bio/SearchIO/BlastIO/blast_tab.py [hunk 5]
```diff
@@ -305,9 +305,7 @@ def _parse_comments(self):
                 self.line = self.line.strip()
 
     def _parse_fields_line(self):
-        """Returns a list of column short names from the 'Fields'
-        comment line.
-        """
+        """Return column short names line from 'Fields' comment line."""
```
Comment (saketkc) [inline Bio/SearchIO/BlastIO/blast_tab.py]: PRIVATE?

File: Bio/SearchIO/_model/hsp.py [hunk 6]
```diff
@@ -792,7 +792,7 @@ def _str_aln(self):
 
     # sequence properties #
     def _set_seq(self, seq, seq_type):
-        """Checks the given sequence for attribute setting
+        """Checks the given sequence for attribute setting.
```
Comment (saketkc) [inline Bio/SearchIO/_model/hsp.py]: PRIVATE?

File: Bio/motifs/jaspar/db.py [hunk 7]
```diff
@@ -422,7 +422,7 @@ def _fetch_motif_by_internal_id(self, int_id):
         return motif
 
     def _fetch_counts_matrix(self, int_id):
-        """Fetch the counts matrix from the JASPAR DB by the internal ID
+        """Fetch the counts matrix from the JASPAR DB by the internal ID.
```
Comment (saketkc) [inline Bio/motifs/jaspar/db.py]: PRIVATE?

File: Bio/PDB/PDBList.py (hunk 2 above)
Comment (peterjc) [reply Bio/PDB/PDBList.py]: Yes, especially since it is so short. Good spot.

File: Bio/Phylo/Consensus.py (hunk 4 above)
Comment (peterjc) [reply Bio/Phylo/Consensus.py]: Ideally, but it is already a very long line.

File: Bio/Sequencing/Applications/_samtools.py (hunk 1 above)
Comment (peterjc) [reply Bio/Sequencing/Applications/_samtools.py]: It does look wrong on GitHub, I agree. Maybe emacs stuck a tab in there? I must check that...

File: Bio/SearchIO/ExonerateIO/_base.py [hunk 8]
```diff
@@ -106,8 +106,9 @@ def _get_fragments_phase(frags):
 
 
 def _adjust_aa_seq(fraglist):
-    """Transforms three-letter amino acid codes into one-letters in the
-    given HSPFragments.
+    """Transform fragment's 3-letter amino acid codes to one-letter (PRIVATE).
```
Comment (bow) [inline Bio/SearchIO/ExonerateIO/_base.py]: Probably `'Transform 3-letter AA codes of input fragments to one-letter codes (PRIVATE).'` is a better description.

**Comment by bow** [comment]
@peterjc , aside from the comments @saketkc has made, I've added just one suggestion. The rest seems fine.

File: Bio/Phylo/NewickIO.py [hunk 9]
```diff
@@ -212,18 +212,14 @@ def _parse_tree(self, text):
...
         clade = Newick.Clade()
         if parent:
             clade.parent = parent
         return clade
 
     def process_clade(self, clade):
-        """Final processing of a parsed clade. Removes the node's parent and
-        returns it.
-        """
+        """Final processing of parsed clade. Removes node's parent and return it."""
```
Comment (etal) [inline Bio/Phylo/NewickIO.py]: "Removes" -> "Remove"

**Comment by etal** [comment]
The Phylo and PDB changes look fine to me.

**Comment by peterjc** [comment]
I've logged #1466 for missing ``(PRIVATE)`` in our private function/method's docstrings.

**Comment by peterjc** [comment]
Thank you all - those extra eyes caught a few things I'd missed.

**Pull request merged**